components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -612,7 +612,8 @@ impl Duration {
 
     fn format(self, sep: &str) -> String {
         use std::fmt::Write;
-        let mut string = String::new();
+        let res_max_len = 1 + 6 + 2 * sep.len() + 1 + MAX_FSP as usize;
```

Comment:
1. If you try to use such expression to explain the length of each part, please add some comments.
2. Otherwise, `sep.len() << 1 + 14`